subprojects/core/src/main/java/org/gradle/configuration/ScriptPluginFactorySelector.java

```diff
@@ -126,4 +132,11 @@ private List<ScriptingLanguage> scriptingLanguages() {
     private ScriptPluginFactory instantiate(String provider) {
         return providerInstantiator.instantiate(provider);
     }
+
+    private void observeRemoteScriptIfPresent(ScriptSource scriptSource) {
+        URI scriptLocationUri = scriptSource.getResource().getLocation().getURI();
+        if (scriptLocationUri != null && scriptLocationUri.getScheme().startsWith("http")) {
+            remoteScriptListener.remoteScriptObserved(scriptSource);
```

Comment:
We can't passing here `scriptSource.resource.location.contentHash` because it's leading to making network call. This not desirable behavior if we have nonCC build invocation and break some tests. So decided that it's reasonable to guard `scriptSource.resource.location.contentHash` by `CCWriter.isInputTrackingDisabled()`